iroha_data_model/src/lib.rs

```diff
@@ -1455,11 +1463,68 @@ pub mod transaction {
         }
     }
 
+    #[derive(Debug, Clone, Serialize, Deserialize, Encode, Decode, PartialEq, Eq)]
+    pub enum FinalizedTransaction {
+        Accepted {
+            transaction: Transaction,
+        },
+        Rejected {
+            transaction: Transaction,
```

Comment:
Also transactions here should be versioned

Reply:
Rejected variant can have `VersionedRejectedTransaction` and Accepted can have `VersionedAcceptedTransaction`

Reply:
It looks better. The problem that structs `Versioned*Transaction` do not declared in `iroha_data_model`, so using it in `iroha_data_model` require moving a lot of piece of code to this module and making them as a part of our data model, so that's may violate design. If @eadventurous agree I can do that.

Reply:
Yeah, you right. Then at least it can be changed to just `VersionedTransaction` here. But in this way:
```rust
pub enum FinalizedTransaction {
    /// Accepted (committed) transaction
    Accepted(VersionedTransaction),
    /// Rejected transaction with reason of rejection
    Rejected(FinalizedRejectedTransaction),
}

pub struct FinalizedRejectedTransaction {
    transaction: VersionedTransaction,
    rejection_reason: TransactionRejectionReason,
}
```

Reply:
Probably they also should be versioned. @eadventurous what do you think?

Reply:
I suggest:
1. To move `RejectedTransaction` from `iroha` crate to `iroha_data_model` crate, as it is not a soft wrapper as other state transaction structs and not to duplicate it here.
2. I think `RejectedTransaction` has also a versioned struct so it should also be moved to data model. 
3. Rename `FinalizedTransaction` to `TransactionValue` to represent that it is not a state but just a value in the `Value` enum.

Otherwise I agree with the structure:
```rust
pub enum TransactionValue {
    Transaction(VersionedTransaction),
    RejectedTransaction(VersionedRejectedTransaction),
}

pub enum Value {
   TransactionValue(..),
   ...
}
```